pkg/migrate/marketplacemigration.go

```diff
@@ -0,0 +1,367 @@
+package migrate
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"github.com/Axway/agent-sdk/pkg/agent/cache"
+	agentcache "github.com/Axway/agent-sdk/pkg/agent/cache"
+	"github.com/Axway/agent-sdk/pkg/apic"
+	v1 "github.com/Axway/agent-sdk/pkg/apic/apiserver/models/api/v1"
+	mv1a "github.com/Axway/agent-sdk/pkg/apic/apiserver/models/management/v1alpha1"
+	"github.com/Axway/agent-sdk/pkg/apic/provisioning"
+	"github.com/Axway/agent-sdk/pkg/config"
+	"github.com/Axway/agent-sdk/pkg/util/log"
+)
+
+// MarketplaceMigrator interface for performing an marketplace provisioning migration
+type MarketplaceMigrator interface {
+	Migrate(ri *v1.ResourceInstance) (*v1.ResourceInstance, error)
+}
+
+// NewMarketplaceMigration - creates a new MarketplaceMigration
+func NewMarketplaceMigration(client client, cfg config.CentralConfig, cache cache.Manager) *MarketplaceMigration {
+	return &MarketplaceMigration{
+		client: client,
+		cfg:    cfg,
+		cache:  cache,
+	}
+}
+
+// MarketplaceMigration - used for migrating attributes to subresource
+type MarketplaceMigration struct {
+	client                  client
+	cfg                     config.CentralConfig
+	cache                   agentcache.Manager
+	accessRequestDefinition *mv1a.AccessRequestDefinition
+}
+
+// Migrate -
+func (m *MarketplaceMigration) Migrate(ri *v1.ResourceInstance) (*v1.ResourceInstance, error) {
+	if ri.Kind != mv1a.APIServiceGVK().Kind {
+		return ri, fmt.Errorf("expected resource instance kind to be api service")
+	}
+
+	log.Debugf("migrating marketplace provisioning for service: %s", ri.Name)
+
+	funcs := []migrateFunc{
+		m.updateInst,
+	}
+
+	errCh := make(chan error, len(funcs))
+	wg := &sync.WaitGroup{}
+
+	for _, f := range funcs {
+		wg.Add(1)
+
+		go func(fun migrateFunc) {
+			defer wg.Done()
+
+			err := fun(ri)
+			errCh <- err
+		}(f)
+	}
+
+	wg.Wait()
+	close(errCh)
+
+	for e := range errCh {
```

Comment:
Looping through the funcs and the error channel might not be needed since you only have one func in your funcs array

Reply:
You right... i had multiple funcs, but not anymore